Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM python:3.8-slim-buster AS base
+LABEL key="Jeremy Brex todo_app"
+
+# Add Curl as not part of slim Buster
+RUN apt-get update -y && apt-get install curl -y
+
+# Install Poetry
+
+ENV POETRY_HOME=/poetry
+ENV PATH=${POETRY_HOME}/bin:${PATH}
+
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+
+# Set the working directory.
+WORKDIR /app
+COPY todo_app /app/todo_app
```

Comment:
You could take better advantage of layer caching by moving this step after poetry install. So you don't need to redo poetry install whenever an app file is changed, only when the poetry files are changes.

Also the dev image doesn't need the files copied in if you are going to use a bind mount anyway

Reply:
done. good point